cpp/src/gandiva/precompiled/arithmetic_ops.cc

```diff
@@ -234,6 +234,25 @@ DIV_FLOAT(float64)
 
 #undef DIV_FLOAT
 
+#define ROUND(TYPE)                                                \
+  FORCE_INLINE                                                     \
+  gdv_##TYPE round_##TYPE(gdv_##TYPE number, gdv_int32 place) {    \
+    if (number == 0) {                                             \
+      return 0;                                                    \
+    }                                                              \
+    if (number < 0) {                                              \
+      return -round_##TYPE(-number, place);                        \
+    }                                                              \
+    return floor(number * pow(10, place) + 0.5) * pow(10, -place); \
```

Comment:
it seems inefficient to use a floating point operations for integer values?  

Also for floats is special hanlding for inf/Nan necessary?

Reply:
Modified this PR to handle only integers and long integers. For floats and doubles, handling for inf/NaN will be necessary, so will be taking it up in another PR.

For this particular case, we need to be dealing with cases only where the precision for rounding is a negative number, and hence pow(10, precision) will never run into overflow or underflow (providing a check if the precision is larger than the number of digits in the number, return 0).

Please let me know if you have any suggestions.

Reply:
My suggestion would be to avoid floating point operations altogether.   I haven't tested this but for pow(10, ...) on integers I think either a lookup table or recursive/squaring would like be more efficient.  Then for rounding I think something like:

(num + (pow(10, -(precision + 1)) * 5) / pow(10, -precision) should work.